Block Directory: Use plugin API for installing & deleting block-plugins

ryelle opened the pull request:
## Description
The `/plugins` endpoints were introduced in #22454, and while `/block-directory/install` and `/block-directory/uninstall` were updated to use those endpoints in PHP, we can actually remove those endpoints and call the `plugins` methods directly. See https://github.com/WordPress/gutenberg/pull/22454#issuecomment-631849871

This PR updates the install flow to call `POST __experimental/plugins`, and the uninstall flow to call `PUT __experimental/plugins/${ plugin }` (to deactivate) and `DELETE __experimental/plugins/${ plugin }` to remove the files.

It also removes the install & uninstall endpoints from `__experimental/block-directory`, and updates all the tests.

## How has this been tested?
The php & js unit tests have been updated. I've also tested the block directory flow in browser (chrome):

- Open a post
- Search for a new block, add it
- It should install & be added to the post content
- In another tab, you can open the plugins screen to see it installed
- Back in the editor, remove the block, and save your post
- The block is silently removed, since it's unused
- Back on the plugins screen, the plugin should be gone

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->

ryelle committed:
Block Directory: Use plugin API for installing & deleting block-plugins

ryelle committed:
Remove the install & uninstall endpoints from the block-directory

ryelle commented on packages/block-directory/src/store/actions.js:
```diff
@@ -66,14 +66,15 @@ export function* installBlockType( block ) {
 			method: 'POST',
 		} );
-		yield addInstalledBlockType( block );
+		yield addInstalledBlockType( { ...block, plugin: response.plugin } );
```
This is a bit of a workaround, since we need the main plugin file for uninstallation (ex, `blissful-buttons/blissful-buttons.php`, or `wp-p5js-block/plugin.php`). I'm not sure if we can/should try adding this to the block directory response (from wp.org), or leave this as-is.

StevenDufresne replied:
I don't think the block directory response could return that info nor should it.

I'm fine with this approach for now, but since `plugin` is a pretty generic property we could run into collisions in the future.

TimothyBJacobs reviewed (approved):
This looks great to me!

One alternative might be to add the `self` link to the `addInstalledBlockType` instead. That way you don't have to manually construct the route, and can just operate off the link.

StevenDufresne commented on packages/block-directory/src/store/actions.js:
```diff
@@ -94,18 +95,19 @@ export function* installBlockType( block ) {
  */
 export function* uninstallBlockType( block ) {
-	const { id } = block;
+	const plugin = block.plugin.replace( '.php', '' );
```
This is somewhat unexpected, is the plugin file the only way to get the slug?

TimothyBJacobs replied:
Right now the plugins controller returns the file. This could be bypassed by using the `self` link. 

I don't know if the controller should return the file ie `my-plugin/init.php` or `my-plugin/init` for the plugin field. The former is the "real" value, but the latter is what will get used in the URL. I'm very much open to thoughts.

StevenDufresne commented:
Ran this locally. Works fine.

ryelle committed:
Set the endpoint for the block by using the self link property

ellatrix commented:
Could this be rebased? Not sure if the test failures are legitimate.

ellatrix commented:
Ok, I rebased it.

tellyworth committed:
Remove erroneous route tests

The install and uninstall routes no longer exist on this controller, as they've moved to the plugin controller.

ellatrix commented:
Looks like there's some e2e test failures.

StevenDufresne committed:
Update e2e install mock url to fix tests.
